spec/shared_examples.rb

```diff
@@ -0,0 +1,54 @@
+RSpec.shared_examples 'oas imported' do
+  include_context :resources
+  include_context :temp_dir
+  include_context :random_name
+
+  # render from template to avoid system_name collision
+  let(:system_name) { "test_openapi_#{random_lowercase_name}_#{Time.now.getutc.to_i}" }
+  let(:oas_resource_path) do
+    oas_template = File.read(File.join(resources_path, 'swagger_template.erb'))
+    renderer = OpenAPITemplateRenderer.new(oas_template)
+    renderer.title = system_name
+    tmp_dir.join('swagger.yaml').tap { |file| file.write(renderer.render) }
+  end
+  let(:expected_methods) do
+    [
+      { 'friendly_name' => 'addPet', 'system_name' => 'addpet' },
+      { 'friendly_name' => 'updatePet', 'system_name' => 'updatepet' },
+      { 'friendly_name' => 'findPetsByStatus', 'system_name' => 'findpetsbystatus' }
+    ]
+  end
+  let(:method_keys) { %w[friendly_name system_name] }
+  let(:expected_mapping_rules) do
+    [
+      { 'pattern' => '/v2/pet', 'http_method' => 'POST', 'delta' => 1 },
+      { 'pattern' => '/v2/pet', 'http_method' => 'PUT', 'delta' => 1 },
+      { 'pattern' => '/v2/pet/findByStatus', 'http_method' => 'GET', 'delta' => 1 }
+    ]
+  end
+  let(:mapping_rule_keys) { %w[pattern http_method delta] }
+
+  it 'methods are created' do
+    expect { subject }.to output.to_stdout
+    expect(expected_methods.size).to be > 0
+    # test Set(service.methods) includes Set(expected_methods)
+    # with a custom identity method for methods
+    expect(
+      ThreeScaleToolbox::Helper.array_difference(expected_methods, service.methods) do |el1, el2|
+        ThreeScaleToolbox::Helper.compare_hashes(el1, el2, method_keys)
+      end
+    ).to be_empty
+  end
+
+  it 'mapping rules are created' do
+    expect { subject }.to output.to_stdout
+    expect(expected_mapping_rules.size).to be > 0
+    # test Set(service.mapping_rules) includes Set(expected_mapping_rules)
+    # with a custom identity method for mapping_rules
+    expect(
+      ThreeScaleToolbox::Helper.array_difference(expected_mapping_rules, service.mapping_rules) do |el1, el2|
```

Comment:
I think it would make sense to extract this into a matcher.

```ruby
expect(expected_mapping_rules).to have_no_difference(service.mapping_rules).comparing_keys(mapping_rule_keys)
```

Reply:
Nice
I would say:

```
expect(expected_mapping_rules).to be_subset_of(service.mapping_rules).comparing_keys(mapping_rule_keys)
```

Reply:
Yep, that looks good!